src/QueryBuilder/ResultViewer/utils.js

```diff
@@ -45,3 +45,11 @@ export const formatValueByDataType = (value, dataType, intl, additionalParams =
       return value || '';
   }
 };
+
+export const findLabelByValue = (options, value) => {
+  // Exceptional case for custom field, that can be identified using that pattern,
+  // that comes from JSONB
+  if (!/^(_custom_field|opt_)/.test(value) || Array.isArray(value)) return value;
```

Comment:
This comment does the opposite of what it says. The first match is negated, so this conditional applies to array-values and values that are _not_ custom fields. A function header describing the expected behavior for different conditions (regular value, custom field value, array value) would be helpful.